src/lib/Flasher.js

```diff
@@ -432,16 +440,20 @@ class Flasher {
       return null;
     }
 
-    if (this._missedChunks.size) {
-      return Promise.resolve();
+    let waitCount: number = 20;
+    while (waitCount > 0) {
+      waitCount -= 1;
+      if (this._missedChunks.size > 0) {
+        logger.info(
```

Comment:
Have you actually built an environment to test this scenario?

I'm interested in the case where you have the low bandwidth. Let's say you have a failure at chunk 1 and a failure at chunk n-1. The time between sending the two chunks is over a half second. In that case this function would exit before it saw the second missed chunk and wouldn't send the second chunk.

---

I think I heard that Particle added a new event from the system firmware that may help with this scenario but I'm not 100% sure.